.changeset/redact-stream-chunk-boundary.md

```
@@ -0,0 +1,5 @@
+---
+'@mastra/core': patch
+---
+
+Fixed RegexFilterProcessor redact strategy to redact matches split across stream chunks, and added a `streamCarryoverSize` option for custom rules whose matches can be longer than the default 128-char window. Fixes #21049
```

markdownlint-cli2 (0.23.2)

[warning] First line in a file should be a top-level heading

(MD041, first-line-heading, first-line-h1)